ssl/ssl_sess.c

```diff
@@ -92,18 +92,24 @@ SSL_SESSION *SSL_SESSION_new(void)
     ss->timeout = 60 * 5 + 4;   /* 5 minute timeout by default */
     ss->time = (unsigned long)time(NULL);
     ss->lock = CRYPTO_THREAD_lock_new();
-    if (ss->lock == NULL) {
+    ss->peer_chain = sk_X509_new_null();
+    if (ss->lock == NULL || ss->peer_chain == NULL) {
         SSLerr(SSL_F_SSL_SESSION_NEW, ERR_R_MALLOC_FAILURE);
-        OPENSSL_free(ss);
-        return NULL;
+        goto err;
     }
 
     if (!CRYPTO_new_ex_data(CRYPTO_EX_INDEX_SSL_SESSION, ss, &ss->ex_data)) {
-        CRYPTO_THREAD_lock_free(ss->lock);
-        OPENSSL_free(ss);
-        return NULL;
+        goto err;
     }
     return ss;
+ err:
+    if (ss) {
```

Comment:
All `goto err` are performed after ss is dereferenced [multiple times]. Hence if (ss) is redundant and becomes unjustified load on branch prediction logic.